pkg/action/action.go

```diff
@@ -17,8 +17,58 @@ import (
 	"github.com/rs/xid"
 )
 
+func InsertMulti(args ...interface{}) error {
+	db := args[0].(*sql.DB)
+	driver := args[1].(drivers.Driver)
+	tableToFieldsMap := args[2].(map[string][]drivers.FieldDescriptor)
+	insertionOrder := args[3].([]string)
+	//func InsertMulti(db *sql.DB, driver drivers.Driver, tableToFieldsMap map[string][]drivers.FieldDescriptor, insertionOrder []string) error {
+	tableFieldValuesMap := make(map[string]map[string]interface{})
+	for _, table := range insertionOrder {
+		if fields, ok := tableToFieldsMap[table]; ok {
+			var f = make([]string, 0, len(fields))
+			var values []interface{}
+			for _, field := range fields {
+				f = append(f, field.Field)
+				if field.HasDefaultValue {
+					continue
+				}
+				var data interface{}
+				if field.ForeignKeyDescriptor != nil {
+					if foreignTableFields, ok := tableFieldValuesMap[field.ForeignKeyDescriptor.ForeignTableName]; ok {
+						if val, ok := foreignTableFields[field.ForeignKeyDescriptor.ForeignColumnName]; ok {
+							data = val
+							continue
+						}
+					}
+					val, err := driver.GetLatestColumnValue(field.ForeignKeyDescriptor.ForeignTableName, field.ForeignKeyDescriptor.ForeignColumnName, db)
+					if err != nil {
+						return err
+					}
+					data = val
+					// Get from table. If no value present in table as well, throw error.
+				} else {
+					data = generateData(driver, field)
+				}
+				values = append(values, data)
+			}
+			query := driver.Insert(f, table)
+			_, err := db.Exec(query, values...)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
 // Insert is inserting a random generated data into the chosen table
-func Insert(db *sql.DB, fields []drivers.FieldDescriptor, driver drivers.Driver, table string) error {
+func Insert(args ...interface{}) error {
```

Comment:
Why did you change this to be that generic?

Reply:
This change was to use same worker pool logic for both insert and multi insert .  But  I also discussed this with @PumpkinSeed and one more solution i can think of is  something like this.
type insertActionStruct struct{
    insertArgs *Insert
    multiInsertArgs *MultiInsert
}

and 
func (ias *insertActionStruct) insert(){
}

and
func runHelper(f flags.Flags, ias insertActionStruct) error {
...
...
}
If there's any other alternative, let me know.

Reply:
What you can do here is creating a struct like that:

```
type InsertEnum struct {
  Single *SingleInsertParams
  Multi   *MultiInsertParams
}
```

If you pass nil MultiInsertParams into the MultiInsert it will throw en error. In this case the programmer will know what parameters should have been passed to the underlying function and we have a kind of type safety.